docs/src/theme/NavbarItem/DocsVersionDropdownNavbarItem.tsx

```diff
@@ -17,25 +50,78 @@ import React from "react";
  *
  * See {@link https://docusaurus.io/docs/swizzling/ | here} for more information on swizzling.
  */
+export default function DocsVersionDropdownNavbarItem({
+	mobile,
+	docsPluginId,
+	dropdownItemsBefore,
+	dropdownItemsAfter,
+	...props
+}: Props): JSX.Element {
+	const { search, hash } = useLocation();
+	const activeDocContext = useActiveDocContext(docsPluginId);
+	const versions = useVersions(docsPluginId);
+	const { savePreferredVersionName } = useDocsPreferredVersion(docsPluginId);
 
-/**
- * {@link DocsVersionDropdownNavbarItemWrapper} component props.
- */
-type Props = WrapperProps<typeof DocsVersionDropdownNavbarItemType>;
+	function versionToAccessibleLink(version: GlobalVersion, index: number): AccessibleLinkProps {
+		const targetDoc = getVersionTargetDoc(version, activeDocContext);
+		return {
+			"label": version.label,
+			"to": `${targetDoc.path}${search}${hash}`,
+			"isActive": () => version === activeDocContext.activeVersion,
+			"onClick": () => savePreferredVersionName(version.name),
+			"aria-label": `Version ${version.label}, item ${index + 1} of ${versions.length}`,
+			"role": "menuitem",
+			"aria-setsize": versions.length,
+			"aria-posinset": index + 1,
+			"className": "version-dropdown__item",
+		};
+	}
+
+	const items: AccessibleLinkProps[] = [
+		...dropdownItemsBefore,
+		...versions.map((version, index) => versionToAccessibleLink(version, index)),
+		...dropdownItemsAfter,
+	];
+
+	const dropdownVersion = useDocsVersionCandidates(docsPluginId)?.[0];
+
+	const dropdownLabel =
+		mobile === true && items.length > 1
+			? translate({
+					id: "theme.navbar.mobileVersionsDropdown.label",
+					message: "Versions",
+					description: "The label for the navbar versions dropdown on mobile view",
+				})
+			: dropdownVersion.label;
+	const dropdownTo =
+		mobile === true && items.length > 1
+			? undefined
+			: getVersionTargetDoc(dropdownVersion, activeDocContext).path;
 
-/**
- * Wraps the default DocsVersionDropdownNavbarItem to omit the drop-down on non-versioned pages.
- *
- * @remarks
- * Suggested workaround for lack of version dropdown customization.
- * See {@link https://github.com/facebook/docusaurus/issues/4389}.
- */
-export default function DocsVersionDropdownNavbarItemWrapper(props: Props): JSX.Element {
 	// Do not display this navbar item if current page is not a doc
-	const activeDocContext = useActiveDocContext(props.docsPluginId);
 	if (!activeDocContext.activeDoc) {
 		return <></>;
 	}
 
-	return <DocsVersionDropdownNavbarItem {...props} />;
+	if (items.length <= 1) {
+		return (
+			<DefaultNavbarItem {...props} mobile={mobile} label={dropdownLabel} to={dropdownTo} />
+		);
+	}
+
+	return (
+		<div className="version-dropdown-wrapper">
+			<DropdownNavbarItem
+				{...props}
+				mobile={mobile}
+				label={dropdownLabel}
+				to={dropdownTo}
+				items={items}
+				aria-label="Select documentation version"
+				role="combobox"
+				aria-haspopup="listbox"
+				aria-expanded="false"
```

Comment:
This shouldn't be hardcoded to false right?

Reply:
Yes, I have removed it, this specific one is for the version dropdown, and that cant be expanded, but I have removed this because docusaurus can handle this without me adding it. If we decide on nested dropdowns like we previously had, this prop would have to be removed